lib/google/cloud/gemserver/cli/server.rb

```diff
@@ -132,25 +136,38 @@ def set_bundle key, gemserver_url
           #
           # @param [String] key The key to be added to the credentials.
           def set_gem_credentials key
-            puts "Updating bundle config. Enter a name for your key (default "\
-              "is \"master-gemserver-key\""
-            key_name = sanitize_name(user_input)
+            key_name = sanitize_name(user_input("Updating bundle config. Enter"\
+              "a name for your key (default is \"master-gemserver-key\""))
             key_name = key_name.empty? == true ? Configuration::DEFAULT_KEY_NAME : key_name
             puts "Updating #{Configuration::CREDS_PATH}"
             FileUtils.touch Configuration::CREDS_PATH
-            run_cmd "echo \":#{key_name}: #{key}\" >> #{Configuration::CREDS_PATH}"
+            keys = YAML.load_file(Configuration::CREDS_PATH) || {}
+            keys[key_name.to_sym] = key
+            File.open(Configuration::CREDS_PATH, 'w') { |f| YAML.dump keys, f }
```

Comment:
As you said, modifying an existing value (while preserving formatting, comments, etc.) is hard, while adding a new value just requires appending to the file so is easy. So we could just punt on the former and still support the latter.